tracer/test/Datadog.Trace.Security.Unit.Tests/FuzzEncoder.cs

```diff
@@ -0,0 +1,192 @@
+
+            for (var i = 0; i < properties; i++)
+            {
+                writer.WritePropertyName(faker.Random.Utf16String(4, 16));
+                WriteValue(writer, depth);
+            }
+
+            writer.WriteEndObject();
+        }
+
+        private void WriteArray(JsonTextWriter writer, int depth)
+        {
+            writer.WriteStartArray();
+
+            var values = _rnd.Next(15);
+
+            for (var i = 0; i < values; i++)
+            {
+                WriteValue(writer, depth);
+            }
+
+            writer.WriteEndArray();
+        }
+
+        private void WriteTerminatingValue(JsonTextWriter writer)
+        {
+            var i = _rnd.Next(TerminatingTypes.Length);
+            var valueType = TerminatingTypes[i];
+            switch (valueType)
+            {
+                case JTokenType.Integer:
+                    writer.WriteValue(faker.Random.Int());
+                    break;
+                case JTokenType.Float:
+                    writer.WriteValue(faker.Random.Float());
+                    break;
+                case JTokenType.String:
+                    writer.WriteValue(faker.Random.Utf16String(0, 100));
+                    break;
+                case JTokenType.Boolean:
+                    writer.WriteValue(faker.Random.Bool());
+                    break;
+                case JTokenType.Null:
+                    writer.WriteNull();
+                    break;
+                case JTokenType.Date:
+                    writer.WriteValue(faker.Date.Between(DateTime.MinValue, DateTime.MaxValue));
+                    break;
+                case JTokenType.Guid:
+                    writer.WriteValue(Guid.NewGuid());
+                    break;
+                case JTokenType.Uri:
+                    writer.WriteValue(faker.Internet.Url());
+                    break;
+                case JTokenType.TimeSpan:
+                    var timeSpan = DateTime.MaxValue - faker.Date.Between(DateTime.MinValue, DateTime.MaxValue);
+                    writer.WriteValue(timeSpan);
+                    break;
+                default:
+                    Assert.False(true);
+                    break;
+            }
+        }
+
+        private void WriteOpeningValue(JsonTextWriter writer, int depth)
+        {
+            depth++;
+
+            var i = _rnd.Next(OpeningTypes.Length);
+            var valueType = OpeningTypes[i];
+            switch (valueType)
+            {
+                case JTokenType.Object:
+                    WriteObject(writer, depth);
+                    break;
+                case JTokenType.Array:
+                    WriteArray(writer, depth);
+                    break;
+                default:
+                    Assert.False(true);
+                    break;
+            }
+        }
+
+        private MemoryStream GenerateJsonMemoryStream()
+        {
+            var memoryStream = new MemoryStream();
```

Comment:
Fixed, but it meant quite a bit of rejigging the code, if you dispose the other two streams, they end up disposing the memory stream, so it can't be reused. Since there are no unmanaged resources calling disposed is not actually that helpful here (although it should avoid some finalizer calls). See dea63efa22f52e2aec7b97a34e6186345ac481eb for the changes.